[4.1] RavenDB-10697 add netcoreapp2.1 target in Raven.Client

iftahbe opened the pull request.

iftahbe committed:
RavenDB-10697 add netcoreapp2.1 target in Raven.Client

iftahbe committed:
RavenDB-10697 expose WebSocketsRemoteCertificateValidationCallback

iftahbe committed:
RavenDB-10697 restore AuthenticationChangesTests.ChangesWithAuthentication

ppekrol commented on src/Raven.Client/Raven.Client.csproj:
```diff
@@ -3,7 +3,7 @@
     <Description>RavenDB Client is the client library for accessing RavenDB</Description>
     <Authors>Hibernating Rhinos</Authors>
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.1;netstandard2.0</TargetFrameworks>
```
Remember to include the Sparrow.dll in NuGet for netcoreapp2.1. In v4.1 you can use the netcoreapp2.1 Sparrow.dll. In v4.0 you need to use the netstandard2.0 one.

iftahbe committed:
RavenDB-10697 share the certificate validation implementation of RequestExecutor

# Conflicts:
#	src/Raven.Client/Documents/DocumentQueryExtensions.cs
#	src/Raven.Client/Documents/LinqExtensions.cs

iftahbe committed:
RavenDB-10697 small fix

iftahbe committed:
RavenDB-10697 fix PR comments

iftahbe committed:
RavenDB-10697 make CallbackTranslator private
